stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/ChatFirebaseMessagingService.kt

```diff
@@ -1,65 +1,34 @@
 package io.getstream.chat.android.client.notifications
 
-import android.app.NotificationChannel
-import android.app.NotificationManager
-import android.os.Build
-import androidx.core.app.NotificationCompat
+import android.util.Log
 import com.google.firebase.messaging.FirebaseMessagingService
 import com.google.firebase.messaging.RemoteMessage
 import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.logger.ChatLogger
-import io.getstream.chat.android.client.notifications.handler.NotificationConfig
-import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
 
 internal class ChatFirebaseMessagingService : FirebaseMessagingService() {
     private val logger = ChatLogger.get("ChatFirebaseMessagingService")
-    private val defaultNotificationConfig = NotificationConfig()
 
     override fun onMessageReceived(remoteMessage: RemoteMessage) {
         logger.logD("onMessageReceived(): $remoteMessage")
-        if (!ChatClient.isValidRemoteMessage(remoteMessage, defaultNotificationConfig)) {
-            return
-        }
-        createSyncNotificationChannel()
-        showForegroundNotification(defaultNotificationConfig.smallIcon)
-        GlobalScope.launch(DispatcherProvider.IO) {
-            if (ChatClient.isInitialized) {
-                ChatClient.instance().onMessageReceived(remoteMessage)
-            }
-            stopForeground(true)
+        try {
+            ChatClient.handleRemoteMessage(remoteMessage)
+        } catch (exception: IllegalStateException) {
+            Log.e(TAG, "Error while handling remote message: ${exception.message}")
```

Comment:
Can't we use logger here?

Reply:
It doesn't make sense to use logger here. Check this comment: https://github.com/GetStream/stream-chat-android/pull/1859#discussion_r632452263